_posts/2016-04-01-new-post.md

```diff
@@ -0,0 +1,912 @@
+as a team. Since we're all [Ember][ember] developers, many comparisons were
+made.
+
+The first time I went through the React Tutorial was over a year ago and, at the
+time, I had lots of feature envy. Fortunately, many of the awesome features that
+React has are now available in Ember-land.
+
+We're going to use Ember to build the same application so we can compare and
+contrast the way that both frameworks use components.
+
+## Getting Started
+
+We're going to use ember-cli to set up our app. This won't provide us with a
+local API server to run like in the source files provided by the React tutorial
+but we'll use another tool to handle that later on.
+
+If you don't already have ember-cli installed, head over to [Installing
+Ember][installing-ember].
+
+To get started, we'll use ember-cli to generate a new Ember app:
+
+```javascript
+$ ember new ember-comment-box
+```
+
+We can verify that our app is set up correctly by running the local ember server
+and viewing the app in our browser [http://localhost:4200](localhost):
+
+```bash
+$ cd ember-comment-box
+$ ember server
+$ open http://localhost:4200
+```
+
+## Recap
+
+The main features of the app are:
+
+- A list of comments.
+- A form for adding new comments.
+- Load comment data from a server.
+- Implement optimistic updates.
+
+Let's take a look at our component hierarchy:
+
+```no-highlight
+- comment-box
+  - comment-list
+    - comment-list-item
+  - comment-form
+```
+
+We're going to have a `comment-box` component that has two child components,
+`comment-list` and `comment-form`. The `comment-list` component will also have
+many child `comment-list-item` components.
+
+--- do a diagram! ---
+
+Let's get to work!
+
+Note: You may have noticed that we're using the name `comment-list-item` instead
+of just `comment`. This is because Ember components require a dash in their name
+in order to avoid conflicts with HTML element names.
+
+## Your First Component
+
+> React is all about modular, composable components
+
+Great! So is Ember.
+
+The first version of our `comment-box` component will simply display some text.
+Since this component does not yet need any custom logic, we only need to create
+a template:
+
+```handlebars
+{{!-- app/templates/components/comment-box.hbs --}}
+
+Hello, world! I am a comment-box.
+```
+
+Now we can use the `comment-box` component inside the main template for our
+application:
+
+```handlebars
+{{!-- app/templates/application.hbs --}}
+
+{{comment-box}}
+```
+
+### Handlebars vs. JSX
+
+Immediately, we've run into one of the biggest differences between writing
+components for Ember and writing components for React.
+
+With React, whether you're a fan of the [JSX syntax][jsx-syntax] or not, you'll
+be using JavaScript to generate the markup for your UI directly inside your
+component. Relying on JavaScript rather than a separate templating language
+allows developers to leverage the power of all of the features of the JavaScript
```

Comment:
"Relying on Javascript rather than a separate templating language allows developers to leverage the power of the Javascript language and all of its' features they are already familiar with."